### PR TITLE
Transformer to sqs worker

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Bringing the Transformer in line with the other services by using SQS as an event bus.

### Who is this change for?

💻 

### Have the following been considered/are they needed?

- [ ] Deployed new versions

### CODE REVIEW

**Comment by kenoir** [comment]
For https://github.com/wellcometrust/platform-api/issues/478

File: common/src/main/scala/uk/ac/wellcome/sqs/SQSWorker.scala [hunk 1]
```diff
@@ -10,17 +10,16 @@ import uk.ac.wellcome.utils.{JsonUtil, TryBackoff}
 import scala.concurrent.Future
 import scala.util.Try
 
-trait SQSWorker extends TryBackoff {
-
-  val sqsReader: SQSReader
-  val actorSystem: ActorSystem
-  val metricsSender: MetricsSender
+abstract class SQSWorker(sqsReader: SQSReader,
```
Comment (kenoir) [inline common/src/main/scala/uk/ac/wellcome/sqs/SQSWorker.scala]: Why not a trait?

Comment (kenoir) [inline common/src/main/scala/uk/ac/wellcome/sqs/SQSWorker.scala]: Ok, I worked it out :)

**Comment by kenoir** [comment]
This is good to go when Travis agrees.

**Comment by kenoir** [comment]
Failing reindexer test.

**Comment by alicefuzier** [comment]
Reindexer test is the flaky test 😠

**Pull request merged**